### PR TITLE
Expose mediaplayer on player

### PR DESCRIPTION
**Pull request by forbesjo**
This PR is to place the dashjs mediaplayer on player.dash.mediaPlayer. The media player will be recreated every time a source is loaded in the video.js player.

### CODE REVIEW

File: src/js/videojs-dash.js [hunk 1]
```diff
@@ -40,13 +41,9 @@ class Html5DashJS {
     let manifestSource = source.src;
     this.keySystemOptions_ = Html5DashJS.buildDashJSProtData(source.keySystemOptions);
 
-    // Save the context after the first initialization for subsequent instances
-    Html5DashJS.context_ = Html5DashJS.context_ || {};
+    player.dash.mediaPlayer = dashjs.MediaPlayer().create();
```
Comment (imbcmdth) [inline src/js/videojs-dash.js]: Should we be deleting `player.dash` on dispose?

Comment (gkatsev) [inline src/js/videojs-dash.js]: yes, we should delete `player.dash` on dispose.

Comment (forbesjo) [inline src/js/videojs-dash.js]: `this.mediaPlayer_.reset()` is necessary when changing sources from dash to something else.

**Comment by imbcmdth** [comment]
LGTM

**Pull request merged**